core/eolearn/tests/test_core_tasks.py

```diff
@@ -199,46 +199,38 @@ def test_remove_fails(patch: EOPatch) -> None:
         RemoveFeatureTask((FeatureType.BBOX, None))(patch)
 
 
-def test_duplicate_feature(patch):
-    mask_data = np.arange(10).reshape(5, 2, 1, 1)
-    feature_name = "MASK1"
-    duplicate_name = "MASK2"
-
-    patch = AddFeatureTask((FeatureType.MASK, feature_name))(patch, mask_data)
-
-    duplicate_task = DuplicateFeatureTask((FeatureType.MASK, feature_name, duplicate_name))
-    patch = duplicate_task(patch)
-
-    assert duplicate_name in patch.mask, "Feature was not duplicated. Name not found."
-    assert id(patch.mask["MASK1"]) == id(patch.mask["MASK2"])
-    assert np.array_equal(
-        patch.mask[duplicate_name], mask_data
-    ), "Feature was not duplicated correctly. Data does not match."
+@pytest.mark.parametrize(
+    "feature_specification",
+    [
+        [(FeatureType.DATA, "bands", "bands2")],
+        [(FeatureType.DATA, "bands", "bands2"), (FeatureType.MASK_TIMELESS, "mask", "mask2")],
+        [(FeatureType.DATA, "bands", f"bands{i}") for i in range(5)],
+    ],
+)
+def test_duplicate_feature(feature_specification: List[FeaturesSpecification], patch: EOPatch) -> None:
+    patch = DuplicateFeatureTask(feature_specification)(patch)
 
-    with pytest.raises(ValueError):
-        # Expected a ValueError when creating an already exising feature.
-        patch = duplicate_task(patch)
+    for f_type, f_name, f_dup_name in feature_specification:
+        assert f_dup_name in patch[f_type]
+        assert id(patch[(f_type, f_name)]) == id(patch[(f_type, f_dup_name)]), "DuplicateFeatureTask used deep copy."
+        assert_array_equal(patch[(f_type, f_name)], patch[(f_type, f_dup_name)])
```

Comment:
this shouldn't be equal, as you checked that it is the same object

Reply:
Do you mean _should_, and that this check is therefore redundant?

Reply:
Updated

Reply:
ah yes, words are hard